src/modules/did/did.processor.ts

```diff
@@ -23,16 +39,30 @@ export class DIDProcessor {
   @Process(ADD_DID_DOC_QUEUE_NAME)
   public async processDIDDocumentAddition(job: Job<string>) {
     this.logger.debug(`processing cache add for ${job.data}`);
-    await this.didService.addCachedDocument(job.data);
+    const doc = await this.didService.addCachedDocument(job.data);
+
+    await this.pinClaims(doc.service.map((s) => s.serviceEndpoint));
   }
 
   @Process(UPDATE_DID_DOC_QUEUE_NAME)
   public async processDIDDocumentRefresh(job: Job<string>) {
     this.logger.debug(`processing cache refresh for ${job.data}`);
-    if (this.configService.get('DID_SYNC_MODE_FULL') === 'true') {
-      await this.didService.addCachedDocument(job.data, true);
+    let doc: DIDDocumentEntity;
+    if (this.config.get('DID_SYNC_MODE_FULL') === 'true') {
+      doc = await this.didService.addCachedDocument(job.data, true);
     } else {
-      await this.didService.incrementalRefreshCachedDocument(job.data);
+      doc = await this.didService.incrementalRefreshCachedDocument(job.data);
+    }
+
+    await this.pinClaims(doc.service.map((s) => s.serviceEndpoint));
+  }
+
+  private async pinClaims(cids: string[]) {
+    for await (const cid of cids) {
+      if (!this.didCluster.isPinned(cid)) {
```

Comment:
Didn't use here Promise.all, because it would make code more havyweight: three Promise.all for `isPinned`, 'get', `save` and `isPinned` along with pin status need to pass cid

Reply:
Ok, sounds good